wordpress-mobile/WordPress-Android
# [Reading Preferences] Add Analytics Tracking

Add analytics tracking (as discussed in p1712099908480889/1712045333.445079-slack-C06PGTY5PKK) to the Reading Preferences feature.

-----

## To Test:

Prerequisites:
- Open Jetpack
- Turn on the `ReaderReadingPreferencesFeatureConfig` in `Debug Settings > Features in Development`
- Check the situations and expected events in the Google Doc mentioned in p1712099908480889/1712045333.445079-slack-C06PGTY5PKK and run those scenarios on the Jetpack Reader.

-----

## Regression Notes

1. Potential unintended areas of impact

    - Reading Preferences Screen behavior changes

2. What I did to test those areas of impact (or what existing automated tests I relied on)

    - Manual testing and existing unit tests for the screen

3. What automated tests I added (or what prevented me from doing so)

    - Several unit tests related to tracking

-----

## PR Submission Checklist:


-----

## Testing Checklist (strike-out the not-applying and unnecessary ones):
